mp/src/game/client/momentum/ui/SettingsPanel/ComparisonsSettingsPage.cpp

```diff
@@ -179,11 +174,6 @@ void ComparisonsSettingsPage::LoadSettings()
     {
         m_pMaxZones->SetText(ConVarRef("mom_comparisons_max_zones").GetString());
```

Comment:
This can actually change to be a CVarTextEntry. I am curious how it handles the fact it's limited to 1-10 but I think the convar does it?

Reply:
Turns out it doesn't really. Values can be entered above/below the bounds but the cvars won't update to that value. 

For now I switched everything over that I can. The functionality for this page's bogus comparison panel requires the cvar to be updated every time the text is changed so I kept the `OnTextChanged` override. 

I believe that `CvarTextEntry` could use some more functionality, such as a flag that determines whether the cvar should be updated when the text is changed or just leave it to the `ApplySettings` override. Also could make the text entry error when a value is outside the cvar's bounds, which it current does not do.

Reply:
I think that would be a worthwhile change to try in a new branch. I would love to remove the "Apply" button in the options panel and have things automatically apply upon changes, which the text entry would have to have its own validation with error state + hover tooltip to show what's wrong, preventing applying until it's fixed.

Reply:
Also, did you test this new cvartextentry and make sure it works? Including going over/under min/max?

Reply:
> 
> 
> Also, did you test this new cvartextentry and make sure it works? Including going over/under min/max?

Yes. When a value under/over the min/max is entered, the cvar is rounded to the min/max. When apply is hit, the text box also rounds to that number, although with a trailing `.00000`.

This behaves differently than 0.8.4 though; the text box never gets updated with the min/max value when a number out of range is entered. The bad number is kept when reopening the settings page as well. The cvar still rounds though.

> 
> I would love to remove the "Apply" button in the options panel and have things automatically apply upon changes

Definitely seems doable!

> 
> 
> I think that would be a worthwhile change to try in a new branch. ... which the text entry would have to have its own validation with error state + hover tooltip to show what's wrong, preventing applying until it's fixed.

Will work on this in a separate branch :)